### PR TITLE
Rewrite base models

### PR DESCRIPTION
**Pull request by alessandropellegrini**
The example models have been rewritten.

My experience with this (it could be that the models are bugged!):
- PCS run sequentially fails a self-check (could it be due to events not handled in FIFO order?)
- PCS run with 1 LP with the parallel algorithm, with any number of threads, hangs.

There is a short fix that I've done to the topology library for hex grids. A hex grid of 1 LP is degenerate, but it's a hex grid anyhow, so a special case was inserted in `FindReceiver()`.

@Piccions could you check? Thanks!

### CODE REVIEW

File: models/pcs/pcs.h [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	unsigned int executed_events; // Total number of events
+
+	simtime_t lvt; // Last executed event was at this simulation time
+
+	double ta; // Current call interarrival frequency for this cell
+
+	unsigned int *channel_state;
+	struct _channel *channels;
+} lp_state_type;
+*/
```
Comment (Piccions) [inline models/pcs/pcs.h:115]: I'm not sure we need this block of commented code :monocle_face:

**Review by Piccions — DISMISSED** [review]
Very good PR, PCS definitely needed some love. Thanks for your work!

File: models/pcs/pcs.h (hunk 1 above)
Comment (alessandropellegrini) [reply models/pcs/pcs.h:115]: No we don't! :+1:

**Review by Piccions — APPROVED** [review]
Now we also have the traffic model ported from USE. Thanks for your work @AdrianoPi

**Pull request merged**